engine/examples/dotnetcore.md

```diff
@@ -1,74 +1,102 @@
-4. The `Dockerfile` assumes that your application is called `aspnetapp`. Change the `Dockerfile` to use the .dll file of your project.
-
-  For Linux Containers:
-
-  ```dockerfile
-  FROM microsoft/aspnetcore:1.1
-  WORKDIR /app
-  COPY published ./
-  ENTRYPOINT ["dotnet", "aspnetapp.dll"]
-  ```
-
-  For Windows Containers:
-
-  ```dockerfile
-  FROM microsoft/dotnet:1.1-runtime-nanoserver
-  WORKDIR /app
-  ENV ASPNETCORE_URLS http://+:80
-  COPY published ./
-  ENTRYPOINT ["dotnet", "aspnetapp.dll"]
-  ```
+- Develop and run your ASP.NET Core apps cross-platform on Windows, MacOS and
+  Linux
+- Great for modern cloud-based apps, such as web apps, IoT apps and mobile
+  backends 
+- ASP.NET Core apps can run on [.NET
+  Core](https://www.microsoft.com/net/core/platform) or on the full [.NET
+  Framework](https://www.microsoft.com/net/framework)
+- Designed to provide an optimized development framework for apps that are
+  deployed to the cloud or run on-premise
+- Modular components with minimal overhead retain flexibility while 
+  constructing
+  your solutions
+
+## Prerequisites
+This example assumes you already have an ASP.NET Core app 
+on your machine. If you are new to ASP.NET you can follow a [simple 
+tutorial](https://www.asp.net/get-started) to initialize a project or clone our [ASP.NET Docker Sample](https://github.com/dotnet/dotnet-docker-samples/tree/master/aspnetapp).
+
+## Create a Dockerfile for an ASP.NET Core application
+1.  Create a `Dockerfile` in your project folder.
+2.  Add the text below to your `Dockerfile` for either Linux or [Windows
+   Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/).
+    The tags below are multi-arch meaning they will pull either Windows or 
+    Linux containers depending on what mode is set in [Docker for 
+Windows](https://docs.docker.com/docker-for-windows/). Read more on [switching containers](https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers).
+3.  The `Dockerfile` assumes that your application is called `aspnetapp`. Change
+   the `Dockerfile` to use the DLL file of your project.
+
+```dockerfile
+FROM microsoft/aspnetcore-build:1.1 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+RUN dotnet publish -c Release -o out
+
+# Build runtime image
+FROM microsoft/aspnetcore:1.1
+WORKDIR /app
+COPY --from=build-env /app/out .
+ENTRYPOINT ["dotnet", "aspnetapp.dll"]
+```
 
-5. To make your build context as small as possible add a [`.dockerignore` file](https://docs.docker.com/engine/reference/builder/#dockerignore-file) to your project folder and copy the following into it.
+4.  To make your build context as small as possible add a [`.dockerignore`
+   file](https://docs.docker.com/engine/reference/builder/#dockerignore-file) 
+   to
+   your project folder and copy the following into it.
 
 ```dockerignore
```

Comment:
You can't just put anything after the three backticks to give the language "hint". I don't think `dockerfile` or `dockerignore` are supported. We aren't using the same highlighter we used to. However, `dockerignore` wasn't even supported on our old highlighter. You should use `conf` instead.